pkg/kubectl/resource_columns.go

```diff
@@ -0,0 +1,551 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubectl
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+type columnDefinition struct {
+	label     string
+	content   interface{}
```

Comment:
I understand that the intent of making this `interface{}` instead of `func(interface{}) string` is to enable the type checking below, but it makes it more difficult to determine correctness when writing to the struct.  How about making this `func(interface{}) string` with another field to indicate the required type?  `requiredType reflect.Type` or some such.  

That would make the interactions more clear.